x-pack/plugins/aiops/public/hooks/use_cases_modal.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { stringHash } from '@kbn/ml-string-hash';
+import { AttachmentType } from '@kbn/cases-plugin/common';
+import { EmbeddableChangePointChartInput } from '../embeddable/embeddable_change_point_chart';
+import { EmbeddableChangePointChartType } from '../embeddable/embeddable_change_point_chart_factory';
+import { useAiopsAppContext } from './use_aiops_app_context';
+
+/**
+ * Returns a callback for opening the cases modal with provided attachment state.
+ */
+export const useCasesModal = <EmbeddableType extends EmbeddableChangePointChartType>(
+  embeddableType: EmbeddableType
+) => {
+  const { cases } = useAiopsAppContext();
+
+  const selectCaseModal = cases?.hooks.useCasesAddToExistingCaseModal();
+
+  return useCallback(
+    (persistableState: Partial<Omit<EmbeddableChangePointChartInput, 'id'>>) => {
+      const persistableStateAttachmentState = {
+        ...persistableState,
+        // Creates unique id based on the input
+        id: stringHash(JSON.stringify(persistableState)).toString(),
+      };
+
+      if (!selectCaseModal) {
+        throw new Error('Cases modal is not available');
+      }
+
+      selectCaseModal.open({
+        getAttachments: () => [
+          {
+            type: AttachmentType.persistableState,
+            persistableStateAttachmentTypeId: embeddableType,
+            // TODO Cases: improve type for persistableStateAttachmentState with io-ts
```

Comment:
Is this todo still need to be addressed for this PR?

Reply:
Removed in https://github.com/elastic/kibana/pull/162796/commits/64548e3013563072ba8536ed2c6cae37e4d9dd70

Reply:
Removed in https://github.com/elastic/kibana/pull/162796/commits/64548e3013563072ba8536ed2c6cae37e4d9dd70